Orange/widgets/data/owpreprocess.py

```diff
@@ -535,6 +536,125 @@ def createinstance(params):
+
+        fixed_rb = gui.appendRadioButton(group=strategy_buttons, label="Fixed:",
+                                         insertInto=box)
+        fixed_rb.setMinimumHeight(25)
+        ibox = gui.indentedBox(widget=fixed_rb, sep=63)
+        k_spin = gui.spin(
+            widget=ibox, master=self, value="_k", minv=1, maxv=10, step=1,
+            label="", controlWidth=60, spinType=int, callback=self.edited.emit,
+            disabled=self._strategy != RandomFeatureSelectEditor.Fixed)
+        self._spins[RandomFeatureSelectEditor.Fixed] = k_spin
+
+        percentage_rb = gui.appendRadioButton(group=strategy_buttons,
+                                              label="Percentage:",
+                                              insertInto=box)
+        ibox = gui.indentedBox(widget=box)
+        p_spin = gui.hSlider(
+            widget=ibox, master=self, value="_p", minValue=0, maxValue=100,
+            step=4, ticks=20, labelFormat="%d %%", callback=self.update_p_spin,
+            disabled=self._strategy != RandomFeatureSelectEditor.Percentage)
+        self._spins[RandomFeatureSelectEditor.Percentage] = p_spin
+
+    def update_strategy(self, edited=True):
+        for strategy, spin in self._spins.items():
+            if self._strategy == strategy:
+                spin.setDisabled(False)
+            else:
+                spin.setDisabled(True)
+
+        if edited:
+            self.edited.emit()
+        else:
+            self.changed.emit()
+
+    def update_p_spin(self):
+        curr_val = self._p
+        prev_val = self._ticks_to_update_slider[1]
+        if (self._ticks_to_update_slider[0] is not None and (prev_val is None or
+            any([curr_val < tick <= prev_val or curr_val >= tick > prev_val
+                for tick in self._ticks_to_update_slider[0]]))):
+            self.edited.emit()
+        self._ticks_to_update_slider[1] = curr_val
+
+    def set_k(self, k):
+        if self._k != k:
+            self._k = k
+            self._spins[RandomFeatureSelectEditor.Fixed].setValue(self._k)
+            self.changed.emit()
+
+    def set_p(self, p):
+        if self._p != p:
+            self._p = p
+            self._spins[RandomFeatureSelectEditor.Percentage].setValue(self._p)
+            self.changed.emit()
+
+    def set_max_num_attributes(self, max_to_set):
+        if self._max_num_attributes != max_to_set:
+            self._max_num_attributes = max_to_set
+            k_spin = self._spins[RandomFeatureSelectEditor.Fixed]
+            p_spin = self._spins[RandomFeatureSelectEditor.Percentage]
+            if self._max_num_attributes is None:
+                k_spin.setMaximum(10)
+                p_spin.setTickInterval(100/5)
+                self._ticks_to_update_slider = [None, None]
+            else:
+                k_spin.setMaximum(self._max_num_attributes)
+                p_spin.setTickInterval(p_spin.maximum() /
+                                       self._max_num_attributes)
```

Comment:
No need to adjust `p_spin`. It has percentage granularity since initialization, and that's that.